extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/MockCommandListener.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.mongodb;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.anyOf;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+import com.mongodb.event.CommandListener;
+import com.mongodb.event.CommandStartedEvent;
+
+public class MockCommandListener implements CommandListener {
+
+    private CommandStartedEvent commandStartedEvent;
+
+    @Override
+    public void commandStarted(CommandStartedEvent startedEvent) {
+        this.commandStartedEvent = startedEvent;
+        assertThat(startedEvent, notNullValue());
+        assertThat(startedEvent.getCommandName(), anyOf(equalTo("listDatabases"), equalTo("endSessions")));
+    }
```

Comment:
Having the insertions here is not realy readable for a test.
Can you instead register the CommandEvent in a static List and assert in the test that the event is present inside the list ?